gwpy/utils/tests/test_decorators.py

```diff
@@ -49,3 +49,30 @@ def myfunc(value):
     with pytest.raises(ValueError) as exc:
         myfunc('test')
     assert 'failed to cast return from myfunc as int: ' in str(exc.value)
+
+
+def test_deprecated_function():
+    """Test that `deprecated_function` works without a message
+
+    (and without any functional brackets)
+    """
+    @decorators.deprecated_function
+    def myfunc(value):
+        return str(value)
+
+    with pytest.warns(DeprecationWarning) as record:
+        myfunc('test')
+    assert len(record) == 1
+    assert "myfunc has been deprecated" in str(record[0].message)
+
+
+def test_deprecated_function_message():
+    """Test that `deprecated_function` works with a message
+    """
+    @decorators.deprecated_function(message="don't use {0.__name__}")
+    def myfunc(value):
+        return str(value)
+
+    with pytest.warns(DeprecationWarning) as record:
+        myfunc('test')
+    assert record[0].message == "don't use myfunc"
```

Comment:
I think this `message` attribute needs to be converted to a string:
```suggestion
    assert str(record[0].message) == "don't use myfunc"
```

Reply:
I could have sworn that worked for me locally, maybe only on the latest greatest pytest version (or python 3.9). Anyways, fixed, thanks for the heads up.